### ISSUE TITLE
NullPointerException in applyAndAnimateRemovals

### ISSUE THREAD

**Issue by venator85** [davideas/FlexibleAdapter#716]
Hi,
from Crashlytics I got the following crash:

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
    at android.os.AsyncTask$3.done(AsyncTask.java:353)
    at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
    at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
    at java.util.concurrent.FutureTask.run(FutureTask.java:271)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
    at java.lang.Thread.run(Thread.java:764)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
    at eu.davidea.flexibleadapter.FlexibleAdapter.applyAndAnimateRemovals(FlexibleAdapter.java:4359)
    at eu.davidea.flexibleadapter.FlexibleAdapter.animateTo(FlexibleAdapter.java:4301)
    at eu.davidea.flexibleadapter.FlexibleAdapter.animateDiff(FlexibleAdapter.java:4277)
    at eu.davidea.flexibleadapter.FlexibleAdapter.access$2000(FlexibleAdapter.java:90)
    at eu.davidea.flexibleadapter.FlexibleAdapter$FilterAsyncTask.doInBackground(FlexibleAdapter.java:5483)
    at eu.davidea.flexibleadapter.FlexibleAdapter$FilterAsyncTask.doInBackground(FlexibleAdapter.java:5442)
    at android.os.AsyncTask$2.call(AsyncTask.java:333)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
    at java.lang.Thread.run(Thread.java:764)
```

I'm using `androidx.recyclerview:recyclerview:1.0.0` and flexible-adapter 5.1.0. minSdkVersion is 21, targetSdkVersion is 28, the crash happened on a OnePlus 6 with Android 8.1.0.

Unfortunately I'm unable to provide steps to reproduce the issue, as I don't understand where it may come from. My app uses several RecyclerViews, all powered by flexible-adapter.

Each fragment using flexible-adapter declares a final adapter:
```val adapter = FlexibleAdapter<AbstractFlexibleItem<*>>(mutableListOf(), null, true)```
with an `OnItemClickListener`.

On item click, the item could be refreshed via `notifyItemChanged(position)`. Items are also frequently refreshed with data coming from a websocket by recreating a full list of `AbstractFlexibleItem` and by calling `updateDataSet(rows, true)` (on the UI thread).

`AbstractFlexibleItem`s delegate their `hashCode` and `equals` to their contained item, which is always a Kotlin data class (`hashCode` and `equals` automatically generated over all fields).

List filtering or sticky headers are not used.

Can you help me? If you need more info don't hesitate to ask ;)
Thanks

**Comment by davideas** [comment]
@venator85 thanks. It will be fixed in the next release.

**Comment by laurimaki** [comment]
Hi! Seems like this NPE issue is still not addressed after the [c2be014](https://github.com/davideas/FlexibleAdapter/commit/c2be014d677f9fda08b0b507ca17dacd1d3fdecf). Its still throwing NPE's. Now at the line [4470 of FlexibleAdapter.applyAndAnimateRemovals()](https://github.com/davideas/FlexibleAdapter/blob/10933e8501a15f6fc9c6ecdfafc4bb8732a58e34/flexible-adapter/src/main/java/eu/davidea/flexibleadapter/FlexibleAdapter.java#L4470).

For the background: I cloned this repository master branch with the commit having hash identifier 10933e8, and built the library as .aar, and included that to our project. This NPE crash at FlexibleAdapter.applyAndAnimateRemovals is top 1 crash for our app. I wished that  [c2be014](https://github.com/davideas/FlexibleAdapter/commit/c2be014d677f9fda08b0b507ca17dacd1d3fdecf) would have fixed this issue, but it didn't. It still happens randomly to users. There doesn't seem to be any connecting pattern based on device model or brand or Android OS version.

Here is stack trace from the crash reporting for the NPE crash: 

`Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done + 318(AsyncTask.java:318)
       at java.util.concurrent.FutureTask.finishCompletion + 354(FutureTask.java:354)
       at java.util.concurrent.FutureTask.setException + 223(FutureTask.java:223)
       at java.util.concurrent.FutureTask.run + 242(FutureTask.java:242)
       at java.util.concurrent.ThreadPoolExecutor.runWorker + 1133(ThreadPoolExecutor.java:1133)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run + 607(ThreadPoolExecutor.java:607)
       at java.lang.Thread.run + 762(Thread.java:762)`

`Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
       at eu.davidea.flexibleadapter.FlexibleAdapter.applyAndAnimateRemovals + 4470(FlexibleAdapter.java:4470)
       at eu.davidea.flexibleadapter.FlexibleAdapter.animateTo + 4404(FlexibleAdapter.java:4404)
       at eu.davidea.flexibleadapter.FlexibleAdapter.animateDiff + 4380(FlexibleAdapter.java:4380)
       at eu.davidea.flexibleadapter.FlexibleAdapter.access$1900 + 90(FlexibleAdapter.java:90)
       at eu.davidea.flexibleadapter.FlexibleAdapter$FilterAsyncTask.doInBackground + 5625(FlexibleAdapter.java:5625)
       at eu.davidea.flexibleadapter.FlexibleAdapter$FilterAsyncTask.doInBackground + 5584(FlexibleAdapter.java:5584)
       at android.os.AsyncTask$2.call + 304(AsyncTask.java:304)
       at java.util.concurrent.FutureTask.run + 237(FutureTask.java:237)
       at java.util.concurrent.ThreadPoolExecutor.runWorker + 1133(ThreadPoolExecutor.java:1133)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run + 607(ThreadPoolExecutor.java:607)
       at java.lang.Thread.run + 762(Thread.java:762)`
